stack/rush-stack/src/logic/BasicTasks.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import * as child_process from 'child_process';
+
+import { BuildContext } from './BuildContext';
+
+export class BasicTasks {
+  /**
+   * Build task: Cleans all the temporary files
+   */
+  public static doClean(buildContext: BuildContext): void {
+    const foldersToClean: string[] = [
+      'temp',
+      'lib',
+      'dist'
+    ];
+
+    for (const folderToClean of foldersToClean) {
+      const fullPath: string = path.join(buildContext.projectFolder, folderToClean);
+      console.log(`[clean]: Cleaning "${fullPath}"`);
+      fsx.emptyDirSync(fullPath);
```

Comment:
>fsx.emptyDirSync(fullPath); [](start = 6, length = 27)

Delete the folder if it's just a clean opteration?

Reply:
Deleting the folder causes a race condition

---
In reply to: [179336093](https://github.com/Microsoft/web-build-tools/pull/602#discussion_r179336093) [](ancestors = 179336093)